### PR TITLE
Updates to dev docs + Recommonmark -> MyST Parser

### PR DESCRIPTION
**Pull request by choldgraf**
This PR does two things:

- Makes some minor updates to the development docs per my comments in https://github.com/jupyterhub/repo2docker/pull/1081 (and some minor edits for clarity)
- Updates to use the MyST Parser, so that we can do some more cross-linking between documentation pages in markdown.

updated tasks page is here: https://repo2docker--1082.org.readthedocs.build/en/1082/contributing/tasks.html

### CODE REVIEW

File: docs/source/contributing/tasks.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ These are some common tasks to be done as a part of developing
 and maintaining repo2docker. If you'd like more guidance for how
 to do these things, reach out in the [JupyterHub Gitter channel](https://gitter.im/jupyterhub/jupyterhub).
 
+:::{admonition} Set up your development environment first!
+:class: tip
+Before attempting most tasks, follow the directions to [set up your development environment](contributing:local-dev).
+
+Though note that many of these tasks to **not** require Docker.
```
Comment (manics) [inline docs/source/contributing/tasks.md:11]:
```suggestion
Though note that many of these tasks do **not** require Docker.
```

File: docs/source/contributing/tasks.md [hunk 2]
```diff
@@ -84,35 +90,26 @@ See the subsections below for more detailed instructions.
    ```
 
    This script will resolve dependencies and write them to the respective `.frozen.yml`
```
Comment (manics) [inline docs/source/contributing/tasks.md:92]:
```suggestion
   This script will resolve dependencies and write them to the respective `.lock`
```

File: docs/source/contributing/tasks.md [hunk 3]
```diff
@@ -84,35 +90,26 @@ See the subsections below for more detailed instructions.
...
-5. Make a pull request; see details below.
+4. Make a Pull Request to the `jupyterhub/repo2docker` repository, with a description
+   of what versions were bumped / what new packages were added and why. If you fix a
+   bug or add new functionality consider adding a new test to prevent the bug from
+   coming back/the feature breaking in the future.
 
-6. Once the pull request is approved (but not yet merged), Update the
+5. Once the pull request is approved (but not yet merged), Update the
    change log (details below) and commit the change log, then update
    the pull request.
```
Comment (manics) [inline docs/source/contributing/tasks.md:111]: I've never done this changelog step, and I've never seen anyone else do it either. Should we remove it?

**Review by manics — COMMENTED** [review]
A few minor suggestions, but LGTM.

File: docs/source/contributing/tasks.md (hunk 3 above)
Comment (choldgraf) [reply docs/source/contributing/tasks.md:111]: +1 from me if it's not being followed

Comment (betatim) [reply docs/source/contributing/tasks.md:111]: I don't think anyone does this and we eventually gave up on having a changelog that is accurate, so let's remove it :(

File: docs/source/contributing/tasks.md [hunk 4]
```diff
@@ -83,36 +89,27 @@ See the subsections below for more detailed instructions.
...
```
Comment (choldgraf) [inline docs/source/contributing/tasks.md:111]:
```suggestion
```

OK this removes it!

**Review by betatim — APPROVED** [review]

**Pull request merged**